uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as HRJobContractFabricator;
+use CRM_HRLeaveAndAbsences_BAO_LeaveBalanceChange as LeaveBalanceChange;
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_Service_LeaveRequest as LeaveRequestService;
+use CRM_HRLeaveAndAbsences_Test_Fabricator_WorkPattern as WorkPatternFabricator;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Service_LeaveRequestTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_Service_LeaveRequestTest extends BaseHeadlessTest {
+
+  public function testCreateAlsoCreateTheLeaveRequestBalanceChanges() {
+    $contact = ContactFabricator::fabricate();
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact['id']],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default']);
+
+    $leaveRequestService = new LeaveRequestService();
+
+    $leaveRequestDateTypes = array_flip(LeaveRequest::buildOptions('from_date_type', 'validate'));
+
+    // a 7 days leave request, from monday to sunday
+    $leaveRequest = $leaveRequestService->create([
+      'type_id' => 1,
+      'contact_id' => $contact['id'],
+      'status_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'from_date_type' => $leaveRequestDateTypes['all_day'],
+      'to_date' => CRM_Utils_Date::processDate('2016-01-07'),
```

Comment:
Yeah, I thought about that (and about adding a lot of tests for other different scenarios) but the problem is that this method is just a wrapper around LeaveRequestBAO.create(). If I add a test like that, I would only be duplicating the BAO's tests, increasing the time to run the tests and not really adding any value. Ideally, the only thing that would have to be tested here is if the service calls the BAO create method with the right data or not. Thanks to Civi that is a static method and cannot be mocked. So, the tests only focus on the things that the service does additionaly to creating the LeaveRequest: creating the balance changes.